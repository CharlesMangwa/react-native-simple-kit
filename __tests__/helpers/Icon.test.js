import '@mocks/SVG'
import 'react-native'
import renderer from 'react-test-renderer'
import getIcon from '@helpers/icon'

describe('Helper function: getIcon', () => {
  it('renders correctly', () => {
    const component = renderer.create(getIcon('backButton', 100))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with activated parameter', () => {
    const component = renderer.create(getIcon('backButton', 100, true))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with color parameter', () => {
    const component = renderer.create(
      getIcon('backButton', 100, false, '#122E46')
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
