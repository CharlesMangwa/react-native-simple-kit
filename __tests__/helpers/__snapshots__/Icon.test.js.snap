// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper function: getIcon renders correctly 1`] = `
<View
  activated={false}
  defaultActivatedColor="#53C000"
  defaultColor="#b8b8b8"
  name="backButton"
  size={100}
>
  <Svg />
</View>
`;

exports[`Helper function: getIcon renders with activated parameter 1`] = `
<View
  activated={true}
  defaultActivatedColor="#53C000"
  defaultColor="#b8b8b8"
  name="backButton"
  size={100}
>
  <Svg />
</View>
`;

exports[`Helper function: getIcon renders with color parameter 1`] = `
<View
  activated={false}
  defaultActivatedColor="#53C000"
  defaultColor="#122E46"
  name="backButton"
  size={100}
>
  <Svg />
</View>
`;
